pytential/symbolic/execution.py

```diff
@@ -462,19 +484,26 @@ def _prepare_expr(places, expr):
     from pytential.source import LayerPotentialSourceBase
     from pytential.symbolic.mappers import (
             ToTargetTagger,
-            DerivativeBinder,
-            InterpolationPreprocessor)
+            DerivativeBinder)
 
-    expr = ToTargetTagger(*places.auto_where)(expr)
+    if auto_where is None:
+        auto_where = places.auto_where
+    if not isinstance(auto_where, tuple):
```

Comment:
How come `GeometryCollection` doesn't perform this normalization, and (maybe) offer `default_source`, `default_target`? (Persisting non-normalized state for a long time just invites inconsistency in how it's treated.)

Reply:
It does. This is called by `bind`, where the user can theoretically give a different `auto_where` to what's in the `GeometryCollection`, so it tries to merge the two with a preference for the user overwriting.

I extracted this into a separate method that's called both here and in the `GeometryCollection` constructor so that everyone gets consistent handling.